install/install.ps1

```diff
@@ -12,83 +12,196 @@ if ([string]::IsNullOrEmpty($version)) {
-Add-Type -AssemblyName System.IO.Compression.FileSystem
-[System.IO.Compression.ZipFile]::ExtractToDirectory($zipFile, $azureauthDirectory)
+    Write-Verbose "Extracting ${zipFile} to ${extractedDirectory}"
+    Add-Type -AssemblyName System.IO.Compression.FileSystem
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($zipFile, $azureauthDirectory)
 
-if (Test-Path -Path $latestDirectory) {
-    Write-Verbose "Removing pre-existing latest directory at ${latestDirectory}"
-    
-    # We use IO.Directory::Delete instead of Remove-Item because on Windows Server 2012 with PowerShell 4.0 the latter will not work.
-    [IO.Directory]::Delete($latestDirectory)
-}
+    if (Test-Path -Path $latestDirectory) {
+        Write-Verbose "Removing pre-existing latest directory at ${latestDirectory}"
+        
+        # We use IO.Directory::Delete instead of Remove-Item because on Windows Server 2012 with PowerShell 4.0 the latter will not work.
+        [IO.Directory]::Delete($latestDirectory)
+    }
 
-# We use a directory junction here because not all Windows users will have permissions to create a symlink.
-# We create this junction with cmd.exe's mklink because it has a stable interface across all active versions of Windows and Windows Server,
-# while PowerShell's New-Item has breaking changes and doesn't have the -Target param in 4.0 (the default PowerShell on Win Server 2012).
+    # We use a directory junction here because not all Windows users will have permissions to create a symlink.
+    # We create this junction with cmd.exe's mklink because it has a stable interface across all active versions of Windows and Windows Server,
+    # while PowerShell's New-Item has breaking changes and doesn't have the -Target param in 4.0 (the default PowerShell on Win Server 2012).
+
+    Write-Verbose "Linking ${latestDirectory} to ${extractedDirectory}"
+    cmd.exe /Q /C "mklink /J `"$latestDirectory`" `"$extractedDirectory`"" > $null
+    if (!$?) {
+        Write-Error "Linking failed!"
+    }
 
-Write-Verbose "Linking ${latestDirectory} to ${extractedDirectory}"
-cmd.exe /Q /C "mklink /J `"$latestDirectory`" `"$extractedDirectory`"" > $null
-if (!$?) {
-    Write-Error "Linking failed!"
+    Write-Verbose "Removing ${zipFile}"
+    Remove-Item -Force $zipFile
+
+    # Permanently add the latest directory to the current user's $PATH (if it's not already there).
+    # Note that this will only take effect when a new terminal is started.
+    $registryPath = 'Registry::HKEY_CURRENT_USER\Environment'
+    $currentPath = (Get-ItemProperty -Path $registryPath -Name PATH -ErrorAction SilentlyContinue).Path
+    if ($currentPath -NotMatch 'AzureAuth') {
+        Write-Verbose "Updating `$env:PATH to include ${latestDirectory}"
+        $newPath = if ($null -eq $currentPath) {
+            "${latestDirectory}"
+        } else {
+            "${currentPath};${latestDirectory}"
+        }
+        setx PATH $newPath > $null

```

Comment:
I do think it's worth commenting on why it's important to use `setx` here, since it handles signaling the env vars update to explorer.ese (the god parent of everything on windows apparently).

Reply:
Done.